src/include/planner/plannodes/abstract_scan_plan_node.h

```diff
@@ -123,7 +123,7 @@ class AbstractScanPlanNode : public AbstractPlanNode {
   /**
    * @return predicate used for performing scan
    */
-  const std::shared_ptr<parser::AbstractExpression> &GetScanPredicate() const { return scan_predicate_; }
+  const std::unique_ptr<parser::AbstractExpression> &GetScanPredicate() const { return scan_predicate_; }
```

Comment:
This should be returning a `std::managed_ptr<parser::AbstractExpression>`

Reply:
Nice